### PR TITLE
Updates to the repo readme

### PR DESCRIPTION
**Pull request by devoncarew**
Updates to the repo readme:
- add a github actions badge to the readme
- add a table describing the packages available in this repo to the readme
- add a script which can regenerate the readme table (to stdout); this is run manually

### CODE REVIEW

**Comment by devoncarew** [comment]
Here's what the readme looks like after this PR:

Screen Shot 2022-05-24 at 5 46 14 PM

**Comment by devoncarew** [comment]
cc @jamesderlin for review

File: tool/readme.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import 'dart:io';
+
+void main(List<String> args) {
+  var packages = Directory('packages')
```
Comment (devoncarew) [inline tool/readme.dart:4]: I think we should add sorting here (by package name).

**Review by jamesderlin — APPROVED** [review]
LGTM. `readme.dart` is... quick-and-dirty, but I agree that it suffices for now.

File: tool/readme.dart (hunk 1 above)
Comment (devoncarew) [reply tool/readme.dart:4]: (updated the PR to sort by package name)

**Pull request merged**